lib/entities/QuoteResponse.ts

```diff
@@ -100,6 +96,60 @@ export class QuoteResponse implements QuoteResponseData {
       request.type === TradeType.EXACT_INPUT
         ? [request.amount, BigNumber.from(data.amountOut ?? 0)]
         : [BigNumber.from(data.amountIn ?? 0), request.amount];
+
+    // permissioned tokens check
+    try {
```

Comment:
maybe consider breaking this out into a smaller function? Bit hard to read right now with all of the branching

Reply:
could have a function which returns a promise conditionally on the isPermissionedToken check + data.filler being set, etc. and just Promise.all for tokenIn + tokenOut